packages/core/admin/admin/src/components/PluginsInitializer.tsx

```diff
@@ -90,17 +87,16 @@ type SetPluginReadyAction = {
 
 type Action = SetPluginReadyAction;
 
-const reducer: React.Reducer<State, Action> = (state = initialState, action: Action): State =>
-  produce(state, (draftState) => {
-    switch (action.type) {
-      case 'SET_PLUGIN_READY': {
-        set(draftState, ['plugins', action.pluginId, 'isReady'], true);
-        break;
-      }
-      default:
-        return draftState;
+const reducer: React.Reducer<State, Action> = (state = initialState, action: Action): State => {
+  switch (action.type) {
+    case 'SET_PLUGIN_READY': {
+      const newState = { ...state };
+      newState.plugins[action.pluginId].isReady = true;
```

Comment:
According to the react docs shouldn't we be spreading at every level since spread is a shallow copy: https://react.dev/reference/react/useReducer#ive-dispatched-an-action-but-the-screen-doesnt-update

```js
return {
    ...state,
    plugins: {
      ...state.plugins,
      [action.pluginId]: {
        ...state.plugins[action.pluginId],
        isReady: true,
      },
    },
  };
```

Reply:
Yeah no worries, I'll handle it from here. Thanks.